### DISCUSSION TITLE
How to keep running `httpd`?

### DISCUSSION THREAD

**Discussion by winsphinx** [MusicPlayerDaemon/MPD#1871] [Q&A] [answered]
When I use httpd on a local client, I must login one the remote server and run `mpc` to activate the `mpd.service`. But what I need is to keep it running on the remote server, and I can get access to the httpd anytime. How can I do?

I tried `systemctl --user enable mpd` or `systemctl --user enable mpd.socket` and `systemctl --user enable mpd.server`, none of them works as I'd like to.

At last I use screen/tmux to run an `mpc` on the server, then detach it to keep mpc running, thus makes mpd.service activated. But this seems not nicely.

**Comment by skidoo23** [comment]
Do you want this? https://wiki.archlinux.org/title/Music_Player_Daemon#System-wide_configuration or that? https://wiki.archlinux.org/title/Music_Player_Daemon/Tips_and_tricks#Music_streaming_with_the_satellite_setup

**Comment by winsphinx** [comment]
Thanks a lot!

I've read them, but they are too heavy for me. I'd like just a simpler way, such as `nohup mpc &`, but not working.  

I realized remote mpc client may activate from server's port 6600 , but http client (browser) could not activate from port 8000.

**Comment by skidoo23** [comment]
Search and your mpd.socket file.  You will find a line `ListenStream=6600` by default socket does not listen to port 8000. Maybe adding a line `ListenStream=8000` helps.

Please provide some details of your setup (client OS,  configured audio output) and intended use with simple non technial words.

**Reply by MaxKellermann** [reply]
No, the httpd output cannot be socket-activated, if that's what you want to do.

I don't really get the actual problem. So @winsphinx wants to run MPD as part of a user session, and then doesn't like that MPD will only start if there's a user session? And running MPD as s system-wide service is "too heavy"? I don't get it.

**Comment by winsphinx** [comment]
> Search and your mpd.socket file. You will find a line `ListenStream=6600` by default socket does not listen to port 8000. Maybe adding a line `ListenStream=8000` helps.

Thanks, on the other way, I've tried `http://server:6600`, but no helps.

So I think it's the truth that httpd outputs cannot activate the listening port, as @MaxKellermann metioned it.

What I wanted is that: I run an mpd server on my remote server, (I can correctly active it by local mpc on any other places, on many OS), and now I want activate it by httpd, so that I can do only with web-browser.  
Now I think it isn't the right case.

**Comment by skidoo23** [comment]
@winsphinx do you want a web based client like https://github.com/jcorporation/myMPD ?

**Comment by winsphinx** [comment]
> a web based client like https://github.com/jcorporation/myMPD ?

I tried it and found that's a great artcraft, which makes control and operate easily. but it's more of a GUI mpd server than a mpc client.

I have a remote mpd server (without any sound device), shipped a lot of music and I want to listen to music on my local PC or my phone. I have two options: install a client like mpc on every device, or use the built-in web browser. 

After hours of consideration, I assume that certain data packages from #6600 activate mpd.socket, then mpd.socket triggers mpd.server. Therefore the httpd over #8000 is only an output method and cannot trigger anything.

So what I want is to bring the remote sound into my local devices. Is there any advice?

**Comment by skidoo23** [comment, marked as answer]
Use myMPD. Enable local playback and it will control your MPD and play http stream on any OS or device with a web browser.

**Reply by winsphinx** [reply]
Thanks a lot!
After well-configured, it works as I desired.

**Comment by winsphinx** [comment]
> Use myMPD. Enable local playback and it will control your MPD and play http stream on any OS or device with a web browser.
